analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/IsPattern.cs

```diff
@@ -73,20 +73,27 @@ private static SymbolicConstraint ConstraintFromConstantPattern(ProgramState sta
         return isReferenceType ? ObjectConstraint.NotNull.ApplyOpposite(falseBranch) : null;    // "obj is 42" => "obj" is NotNull and obj.ToString() is safe. We don't have this for bool.
     }
 
-    private static ObjectConstraint ConstraintFromRecursivePattern(IRecursivePatternOperationWrapper recursive, bool falseBranch) =>
-        recursive.InputType.IsReferenceType
-            ? falseBranch switch
-            {
-                true => RecursivePatternAlwaysMatchesAnyNotNull(recursive) ? ObjectConstraint.Null : null,
-                _ => ObjectConstraint.NotNull
-            }
-            : null;
+    private static ObjectConstraint ConstraintFromRecursivePattern(IRecursivePatternOperationWrapper recursive, bool falseBranch, SymbolicValue existingValue)
+    {
+        if ((existingValue?.HasConstraint<ObjectConstraint>() ?? false) || !recursive.InputType.IsReferenceType)
```

Comment:
I preffer `is true` or `== true` for better readability.
```suggestion
        if ((existingValue?.HasConstraint<ObjectConstraint>() == true) || !recursive.InputType.IsReferenceType)
```